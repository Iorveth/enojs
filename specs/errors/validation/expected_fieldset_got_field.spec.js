const eno = require('../../../eno.js');

const input = `
languages:
| eno (eno notation)
| json (JavaScript Object Notation)
`.trim();

describe('validation.expectedFieldsetGotField', () => {
  const document = eno.parse(input);

  let error;
  try {
    document.fieldset('languages');
  } catch(err) {
    error = err;
  }

  it(`provides a correct message`, () => {
    expect(error.message).toMatchSnapshot();
  });

  it(`provides correct selection metadata`, () => {
    expect(error.selection).toMatchSnapshot();
  });
});
