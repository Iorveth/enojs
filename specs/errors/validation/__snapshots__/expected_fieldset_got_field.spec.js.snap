// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation.expectedFieldsetGotField provides a correct message 1`] = `
"Instead of the expected fieldset 'languages' a field with this name was found.

   Line | Content
 >    1 | languages:
 >    2 | | eno (eno notation)
 >    3 | | json (JavaScript Object Notation)
"
`;

exports[`validation.expectedFieldsetGotField provides correct selection metadata 1`] = `
Array [
  Array [
    0,
    0,
  ],
  Array [
    2,
    35,
  ],
]
`;
